### PR TITLE
LogStream: Remove timestamp machinery and memory_consumption

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #4704.

Do not merge yet - needs a testsuite run.

### CODE REVIEW

File: doc/news/changes/incompatibilities/20170810MatthiasMaier [hunk 1]
```diff
@@ -1,8 +1,14 @@
-Removed: LogStream: The reproducible test functionality has been removed
-from deallog. This includes <code>test_mode()</code>,
-<code>threshold_double()</code>, and <code>threshold_float()</code>. Use
-numdiff, or other tools that can do output comparison with small numerical
-differences instead. Further, the <code>log_cerr()</code> function has been
-removed.
+Removed: A number of old functionality has been removed from LogStream:
```
Comment (bangerth) [inline doc/news/changes/incompatibilities/20170810MatthiasMaier]: "old functionality" is not countable. -> "Some old functionality" or "A significant amount of old functionality"

File: cmake/checks/check_02_system_features.cmake [hunk 2]
```diff
@@ -42,9 +40,6 @@ CHECK_INCLUDE_FILE_CXX("sys/resource.h" DEAL_II_HAVE_SYS_RESOURCE_H)
 
 CHECK_INCLUDE_FILE_CXX("sys/time.h" DEAL_II_HAVE_SYS_TIME_H)
 
-CHECK_INCLUDE_FILE_CXX("sys/times.h" DEAL_II_HAVE_SYS_TIMES_H)
-CHECK_CXX_SYMBOL_EXISTS("times" "sys/times.h" DEAL_II_HAVE_TIMES)
```
Comment (bangerth) [inline cmake/checks/check_02_system_features.cmake]: In principle, other parts of our code base or that of user codes may be using these `#define`s. I don't think we need to worry too much about that.

File: tests/integrators/cochain_01.cc [hunk 3]
```diff
@@ -296,13 +296,7 @@ void run3d (unsigned int degree)
 
 int main()
 {
-  const std::string logname = "output";
-  std::ofstream logfile(logname.c_str());
-  deallog.attach(logfile);
-  deallog.log_execution_time(false);
-  if (!debugging)
-    {
-    }
```
Comment (bangerth) [inline tests/integrators/cochain_01.cc]: oh, that looks like a really useful 3 lines!

File: tests/integrators/cochain_01.cc [hunk 4]
```diff
@@ -296,13 +296,7 @@ void run3d (unsigned int degree)
...
+  initlog()
```
Comment (tjhei) [inline tests/integrators/cochain_01.cc]: missing ";"?

File: cmake/checks/check_02_system_features.cmake (hunk 2 above)
Comment (tamiko) [inline cmake/checks/check_02_system_features.cmake]: Internally, these two definitions are not used any more. I hope that user code doesn't overly depend on it.

**Comment by tamiko** [comment]
Updated. Let's wait for a full testsuite run to see whether switching `cerr` -> `cout` did break something.

**Comment by tamiko** [comment]
Passes the testsuite, http://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=10062

**Pull request merged**